### PR TITLE
[Feat] Product - Product table refactor

### PR DESCRIPTION
**Pull request by dmendie**
# Description

Update product entity and create migration for the newly added properties. This is to satisfy the database schema design for the product feature

Required by [#91](https://github.com/hngprojects/hng_boilerplate_csharp_web/issues/91), [#89](https://github.com/hngprojects/hng_boilerplate_csharp_web/issues/89), [#88](https://github.com/hngprojects/hng_boilerplate_csharp_web/issues/88), [86](https://github.com/hngprojects/hng_boilerplate_csharp_web/issues/86), [84](https://github.com/hngprojects/hng_boilerplate_csharp_web/issues/84)

# Changes proposed
Adding the following properties Price, Category, CreatedAt, UpdatedAt

## What were you told to do?

Update the Product entity to match the database schema for API design as required by team members

## What did you do?

Added the following properties

- Price
- Category
- CreatedAt
- UpdatedAt

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] Update product entity with the required fields

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

### CODE REVIEW

File: src/Hng.Domain/Entities/Product.cs [hunk 1]
```diff
@@ -2,8 +2,19 @@
...
+        CreatedAt = DateTime.UtcNow;
+    }
+
     public string Name { get; set; }
     public string Description { get; set; }
+    public string[] Category { get; set; }
+    public decimal Price { get; set; }
     public Guid UserId { get; set; }
     public User User { get; set; }
+
```
Comment (Aniyikaye-Alabi) [inline src/Hng.Domain/Entities/Product.cs:17]: Remove this space, it unnecessary.

File: src/Hng.Domain/Entities/Product.cs [hunk 2]
```diff
@@ -2,8 +2,18 @@
 namespace Hng.Domain.Entities;
 public class Product : EntityBase
 {
+    public Product()
+    {
+        Id = Guid.NewGuid();
```
Comment (NdubuisiJr) [inline src/Hng.Domain/Entities/Product.cs:7]: This `Id = Guid.NewGuid();` is okay but I personally prefer the database generates it.

Comment (dmendie) [reply src/Hng.Domain/Entities/Product.cs]: Id initialization removed

File: src/Hng.Web/Hng.Web.csproj [hunk 3]
```diff
@@ -43,8 +43,8 @@
     <ProjectReference Include="..\Hng.Infrastructure\Hng.Infrastructure.csproj" />
   </ItemGroup>
 
-  <Target Name="PostBuild" AfterTargets="PostBuildEvent">
+  <!--<Target Name="PostBuild" AfterTargets="PostBuildEvent">
```
Comment (NdubuisiJr) [inline src/Hng.Web/Hng.Web.csproj:46]: Why did you touch this file?

File: src/Hng.Web/Controllers/ProductController.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Hng.Web.Controllers
+{
```
Comment (NdubuisiJr) [inline src/Hng.Web/Controllers/ProductController.cs:5]: There's no need for a controller if there's no action

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]
You have merge conflicts. Please pull from dev branch and fix all merge conflicts. Ask your team mates for help if you need one.

Also, resolve all the comments below.

File: src/Hng.Web/Controllers/ProductController.cs (hunk 4 above)
Comment (dmendie) [reply src/Hng.Web/Controllers/ProductController.cs:5]: Was just setting up a controller for my other teammates towork in

File: src/Hng.Web/Hng.Web.csproj (hunk 3 above)
Comment (dmendie) [reply src/Hng.Web/Hng.Web.csproj]: restored

**Review by NdubuisiJr — APPROVED** [review]

**Pull request merged**